protocol/cmd/dydxprotocold/main.go

```diff
@@ -23,3 +23,13 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+var tempDir = func() string {
+	dir, err := os.MkdirTemp("", "dydxprotocol")
+	if err != nil {
+		dir = app.DefaultNodeHome
+	}
+	defer os.RemoveAll(dir)
```

Comment:
won't defer delete `dir` when function is returning and this would clear the `DefaultNodeHome` from any existing state if creating a temp dir creation is not possible?

Reply:
Yeah, I just copied this straight from cosmos. But it doesn't matter because the directories are recreated created in `ReadFromClientConfig`.